compiler/noirc_driver/src/lib.rs

```diff
@@ -136,6 +155,17 @@ pub fn prepare_crate(context: &mut Context, file_name: &Path) -> CrateId {
     root_crate_id
 }
 
+pub fn link_to_debug_crate(context: &mut Context, root_crate_id: CrateId) {
+    let path_to_debug_lib_file = Path::new(DEBUG_CRATE_NAME).join("lib.nr");
+    let debug_file_id = context
+        .file_manager
+        .name_to_id(path_to_debug_lib_file)
+        .expect("debug source is expected to be present in file manager");
+    let debug_crate_id = context.crate_graph.add_crate(debug_file_id);
+
+    add_dep(context, root_crate_id, debug_crate_id, DEBUG_CRATE_NAME.parse().unwrap());
```

Comment:
Looks like this can be replaced with a call to `prepare_dependency` and `add_dep` to avoid duplication.

Reply:
That will also link the debug crate with stdlib. I guess it doesn't hurt. I'll try replacing it.